docs/fsharp-cheatsheet.md

```diff
@@ -740,6 +743,112 @@ Another way of implementing interfaces is to use *object expressions*.
 
 *Partial active patterns* share the syntax of parameterized patterns but their active recognizers accept only one argument.
 
+<div id="asynchronous-programming"></div>
+
+## Asynchronous Programming
+
+F# asynchronous programming is centered around two core concepts: async computations and tasks.
+
+    async {
+        do! Async.Sleep (waitInSeconds * 1000)
+        let! asyncResult = asyncComputation
+        use! disposableResult = iDisposableAsyncComputation
+    }
+
+### Async vs Tasks
+
+An async computation is a unit of work, and a task is a promise of a result. A subtle but important distinction.
+Async computations are composable and are not started until explicitly requested; Tasks (when created using the `task` expression) are _hot_:
+
+    let runAsync waitInSeconds =
+        async {
+            printfn "Created Async"
+            do! Async.Sleep (waitInSeconds * 1000)
+            printfn $"Completed Async"
+        }
+        
+    let runTask waitInSeconds =
+        task {
+            printfn "Started Task"
+            do! System.Threading.Tasks.Task.Delay (waitInSeconds * 1000)
+            printfn $"Completed Task"
+        }
+    
+    let asyncComputation = runAsync 5 // returns Async<unit> and does not print anything
+    let newTask = runTask 3  // returns System.Threading.Tasks.Task<unit> and outputs: "Started Task" <3 second delay> "Completed Task"
+
+    asyncComputation |> Async.RunSynchronously  // this now runs the async computation
+    newTask.Wait() // this will have already completed by this point
+
+    // Output:
+    // Started Task
+    // Created Async
+    // Completed Task
+    // Completed Async
+
+### Async and Task Interop
+
+As F# sits in .NET, and a lot of the codebase uses the C# async/await, the majority of actions are going to be executed and tracked using `System.Threading.Tasks.Task<'T>`s.
+
+#### Async.AwaitTask
+
+This converts a Task into an async computation. It has the [signature](#functions-signatures): `Task<'T>` -> `Async<'T>`
```

Comment:
While this is not untrue, it's also not very useful information - there's lots of discussion as to why in this PR, and we landed on this wording:
https://github.com/fsprojects/FSharp.Control.TaskSeq/pull/249/files#diff-0d723d41a6a5de091170b77b03f31ca3954f6a038be6d72fe42f00bd6f19d6f2R96

Reply:
You don't 'convert' a Task to an Async like you 'convert' an int to a string
The thing Abel and I ended up on is also not useful here
You want to convey that this maps the 'handle for a running task' / Promise (or whatever you're going to term a typically running hot thing) into the Async world.
I think I'd keep ti simple though and just say what you use it for, which is to `await` the result of something that you obtai/request/start/are handed as a Task, and lets you consume/await the result in an `async` block.

Speaking of _block_, I think _workflow_ can be a useful term for as `async` block (as opposed to trying to say one is an async and another is a task as the builder vs promise distinction means they are not really comparable directly)

Reply:
I also think I could not mention it without talking about AwaitTaskCorrect and/or AggregateException... :(

Reply:
"Allows creating an async computation from Tasks" ?